build/lib/src/builder/build_step_impl.dart

```diff
@@ -80,7 +80,8 @@ class BuildStepImpl implements BuildStep {
   @override
   Future writeAsBytes(AssetId id, FutureOr<List<int>> bytes) {
     _checkOutput(id);
-    var done = _futureOrWrite(bytes, (b) => _writer.writeAsBytes(id, b));
+    var done =
+        _futureOrWrite<List<int>>(bytes, (b) => _writer.writeAsBytes(id, b));
```

Comment:
would adding `int` to the arg for the function literal fix it? (that seems more desirable to me)